src/DOMNodeComparator.php

```diff
@@ -70,8 +70,15 @@ public function assertEquals($expected, $actual, $delta = 0.0, $canonicalize = f
     private function nodeToText(DOMNode $node, bool $canonicalize, bool $ignoreCase): string
     {
         if ($canonicalize) {
+            /** @psalm-var string|false $c14n */
+            $c14n = @$node->C14N();
+
+            if ($c14n === false) { // try node-to-text without canonicalize
+                return $this->nodeToText($node, !$canonicalize, $ignoreCase);
+            }
+
             $document = new DOMDocument;
-            @$document->loadXML($node->C14N());
+            @$document->loadXML($c14n);
```

Comment:
Are we sure the `loadXML()` call will now work? We're still using error suppression here, so what happens if the the re-parsing fails? Can it even fail?

Reply:
> Are we sure the loadXML() call will now work?

(TIL) It can fail since `C14N()` can also return an empty string if the `$node` is not included in a document.
<https://www.php.net/manual/es/domnode.c14n.php#116871>

> We're still using error suppression here, so what happens if the the re-parsing fails? Can it even fail?

Yes, it can fail when `C14N()` returns an empty string.

Should I verify that `C14N()` does not return `null` and also does not return an empty string?
That would lead to perform comparison without canonicalization on both scenarios.

Reply:
On latest commits, when `c14n()` I check also for empty string and therefor there is no reason for error suppression on `loadXML`.